core/src/main/java/org/apache/iceberg/FloatFieldMetrics.java

```diff
@@ -19,16 +19,14 @@
 
 package org.apache.iceberg;
 
-import java.nio.ByteBuffer;
-
 /**
  * Iceberg internally tracked field level metrics, used by Parquet and ORC writers only.
  * <p>
  * Parquet/ORC keeps track of most metrics in file statistics, and only NaN counter is actually tracked by writers.
  * This wrapper ensures that metrics not being updated by those writers will not be incorrectly used, by throwing
  * exceptions when they are accessed.
  */
-public class FloatFieldMetrics extends FieldMetrics {
+public class FloatFieldMetrics extends FieldMetrics<Object> {
```

Comment:
Why isn't this `FieldMetrics<Float>`?

Reply:
Sorry should have double check on this. Although I guess `Number` might fit better since Double writer will also use this class, and `Double` doesn't extend `Float` (although it really shouldn't cause a difference in execution)?

Reply:
Number works for me.